### DISCUSSION TITLE
Include guard issue - include guards are being created based on the file name but don't include anything about the path, creating conflicts when using two files with the same name [13730]

### DISCUSSION THREAD

**Discussion by tnhjhu** [eProsima/Fast-DDS-Gen#239] [Ideas]
We have two IDLs with the same name but different namespaces:

e.g.  
one_project/a_folder/SomeThing.idl
one_project/b_folder/SomeThing.idl

Using the FastDDSGen tool we have seen that the generated code is actually creating matching include guards because the include guards are just based on the file name.

Can this be patched to generate the include guards using the folder the IDL is in or some additional information to distinguish between two IDLs with matching names and different contents?

**Comment by ForceConstant** [comment]
Not helpful, but I ran into this issue today. I think an easy fix in fast-dds-gen would be to use the '#pragma once' instead in the headers for guards.

**Reply by JSG-0** [reply]
"pragma once" is not standard (though it is supported in  major compilers)

**Comment by tparys** [comment]
Found this issue after submitting a PR. Agreed that #pragma once might be cleaner provided it is available on supported compilers.

**Comment by JLBuenoLopez** [comment]
Labeling this ticket as enhancement and moving it to the Ideas discussion forum where enhancements and feature requests are discussed.

**Comment by jwillemsen** [comment]
With our code generator we add a hash to the include guard, see for example https://github.com/RemedyIT/idl2cpp11/blob/de2b1cd7ab5951cc856179a9030297c2c5e8806d/examples/const/testC.h#L9
